### ISSUE TITLE
Encoding::CompatibilityError: incompatible character encodings: UTF-8 and ASCII-8BIT

### ISSUE THREAD

**Issue by dandan** [trusche/httplog#23]
Under certain circumstances, a Encoding::CompatibilityError is raised (see backgrace below).  This seems to occur only:

* On 0.2.11 (does not occur on v 0.2.10)
* When a request contains a specific mix of characters.  In my case I'm sending a request to elastic search which contains both a non ascii character, and an escaped URL: "é %A9"

Backtrace:
```
Encoding::CompatibilityError: incompatible character encodings: UTF-8 and ASCII-8BIT
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.2.4/lib/active_support/core_ext/uri.rb:15:in `gsub'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.2.4/lib/active_support/core_ext/uri.rb:15:in `unescape'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/2.2.0/uri/common.rb:125:in `unescape'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/httplog-0.2.11/lib/httplog/http_log.rb:96:in `log_data'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/httplog-0.2.11/lib/httplog/adapters/net_http.rb:17:in `request'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/honeybadger-2.0.12/lib/honeybadger/plugins/net_http.rb:21:in `block in request_with_honeybadger'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.2.4/lib/active_support/notifications.rb:164:in `block in instrument'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.2.4/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/activesupport-4.2.4/lib/active_support/notifications.rb:164:in `instrument'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/honeybadger-2.0.12/lib/honeybadger/plugins/net_http.rb:20:in `request_with_honeybadger'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/2.2.0/net/http.rb:1376:in `block in request'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/2.2.0/net/http.rb:853:in `start'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/2.2.0/net/http.rb:1374:in `request'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/httplog-0.2.11/lib/httplog/adapters/net_http.rb:21:in `block in request'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
	from /home/daniel/.rbenv/versions/2.2.0/lib/ruby/gems/2.2.0/gems/httplog-0.2.11/lib/httplog/adapters/net_http.rb:20:in `request'
```

**Comment by trusche** [comment]
Thanks for reporting this. A bit strange that it should affect only 0.2.10, but in any case, I suspect #22 will fix this. I'll hope to have time to take a closer look at this in the next few days.

**Comment by trusche** [comment]
Oh, it's probably the introduction of `unescape` in this commit: https://github.com/trusche/httplog/commit/bddf6cc2582c7d0cc24ef5236a8535b0e4a94e1a. I may have to revert that or at least make it optional. Now that I think about it, though it might help with readability, it does obfuscate the actual data going over the wire, so maybe it shouldn't be there at all.

**Comment by trusche** [comment]
@dandan in the meantime, there's no new functionality other than that in 0.2.11, it's mostly just gem version bumps, so just fix to 0.2.10 until this is sorted please if that works for you.

**Comment by trusche** [comment]
@dandan could you try using the master branch and let me know if it fixes your issue please?

**Comment by ryansouza** [comment]
@dandan we were running in to this exact backtrace too and https://github.com/trusche/httplog/commit/26edec8dd98bb61b2b2169f290d16930d4da47e6 (released as 0.2.13) has fixed it for us
